src/lambda_extract_metrics.py

```diff
@@ -58,7 +40,10 @@ def process_individual_resource(
     # 1. Create an extractor object for a specific service
     metrics_extractor = MetricsExtractorProvider.get_metrics_extractor(
         resource_type=resource_type,
-        aws_service_client=aws_service_client,
+        account_id=account_id,
```

Comment:
I know we need 2 boto3 clients to get lambda metrics, but I'm thinking of ways other than passing 4 arguments all the way down thru layer.

Just an idea - let's consider, maybe, it can be a bit more estethically pleasing:

create lib.aws class Boto3ClientCreator
__init__ (account_id, region, role_name)
self.get_client(service_name)

and we can pass just client_creator var instead of all these.
Makes sense?

Reply:
Thanks for the commet, totally makes sense, done :)